src/dirs.proj

```diff
@@ -19,11 +19,44 @@
     <Project Include="XMakeBuildEngine\UnitTestsPublicOM\Microsoft.Build.Engine.OM.UnitTests.csproj" />
   </ItemGroup>
 
+  <Import Project="dir.targets" />
   <Import Project="..\dir.traversal.targets" />
 
   <PropertyGroup>
+    <SkipVersionGeneration>true</SkipVersionGeneration>
+    <!-- Turn off the automated package generation in packages.targets:AddNuGetPackageVersionMetadataToNuspecs;
+         we'll use the one produced by NerdBank.GitVersioning instead. -->
+    <DoNotGeneratePackageVersion>true</DoNotGeneratePackageVersion>
+  </PropertyGroup>
+
+  <Import Project="$(ToolsDir)packages.targets" Condition="Exists('$(ToolsDir)packages.targets') and '$(ImportGetNuGetPackageVersions)' != 'false'" />
+
+  <PropertyGroup Condition="Exists('$(ToolsDir)packages.targets') and '$(ImportGetNuGetPackageVersions)' != 'false'">
+    <TraversalBuildDependsOn>
+      $(TraversalBuildDependsOn);
+      GetNuspecProperties;
+      BuildPackages;
+    </TraversalBuildDependsOn>
+    <!-- We use Microsoft.DotNet.Build tools to create our packages, but we want the version comes from 
+         NerdBack.GitVersioning. -->
+    <GetNuGetPackageVersionsDependsOn>SetPackageVersion</GetNuGetPackageVersionsDependsOn>
+  </PropertyGroup>
+
+  <PropertyGroup>
     <!-- Explicitly set the OutDir as it is used by the packaging targets -->
     <OutDir>$(BaseOutputPathWithConfig)</OutDir>
   </PropertyGroup>
 
+  <Target Name="SetPackageVersion" DependsOnTargets="GetNuGetPackageVersion" Returns="$(PackageVersion)">
```

Comment:
(tl;dr: remove it here)

Well, this is one of our dark corners.

Mostly, you don't need a return value from a target: you're running it for its side effects, either on disk (from running tasks that produce files or change the world's state), or in the project (by changing properties and items in memory). But sometimes you do need a concept of a return value: when you're asking another project "what's your output file?", for example.

MSBuild's behavior is that the return value of a target defaults to its `Outputs`, but that can be overridden by adding a `Returns` attribute. That's useful in two main situations:
1. You have multiple incidental outputs, but one "main" output (maybe you produce debug symbols and want to rerun if they're stale, but your real output is only a DLL).
1. You want to communicate something that's not a file-on-disk output to another project--like a computed property.

I don't think you're in either case here, so there shouldn't be a need to have the `Returns`.